clients/imodels-client-management/src/operations/imodel/IModelOperations.ts

```diff
@@ -207,28 +254,69 @@ export class IModelOperations<TOptions extends OperationOptions> extends Operati
     };
   }
 
+  private async isIModelInitialized(params: {
+    authorization: AuthorizationCallback;
+    iModelId: string;
+    errorCodeOnFailure: IModelsErrorCode;
+    headers?: HeaderFactories;
+  }): Promise<boolean> {
+    const { state } = await this._iModelsClient.operations.getCreateIModelDetails({
+      authorization: params.authorization,
+      iModelId: params.iModelId,
+      headers: params.headers
+    });
+
+    if (state !== IModelCreationState.Scheduled &&
```

Comment:
would be easier to check this in all - templated, baseline and cloned - iModel cases
https://developer.bentley.com/apis/imodels-v2/operations/get-imodel-details/
![image](https://github.com/iTwin/imodels-clients/assets/70565417/57cbe47a-438c-49ef-b65e-31d745630929)

Reply:
The returned state can only be `initialized` or `notInitialized`, therefore we cannot know if it failed.